### PR TITLE
Rename keys like `LAlt` to `AltLeft`

### PR DESCRIPTION
**Pull request by jim-ec**
# Objective

The [`KeyCode`](https://github.com/bevyengine/bevy/blob/main/crates/bevy_input/src/keyboard.rs#L86) enum cases `LWin` and `RWin` are too opinionated because they are also assigned meaning by non-Windows operating systems. macOS calls the keys completely different.

## Solution

Match [winits approach](https://github.com/rust-windowing/winit/blob/master/src/keyboard.rs#L1635) naming convention.

---

## Migration Guide

Migrate by replacing:
- `LAlt` → `AltLeft`
- `RAlt` → `AltRight`
- `LBracket` → `BracketLeft`
- `RBracket` → `BracketRight`
- `LControl` → `ControlLeft`
- `RControl` → `ControlRight`
- `LShift` → `ShiftLeft`
- `RShift` → `ShiftRight`
- `LWin` → `SuperLeft`
- `RWin` → `SuperRight`

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
LGTM, but if possible we should add a doc alias to the enum variant. No idea if that's supported though :)

**Comment by CGMossa** [comment]
This needs a little more careful consideration.

**Comment by mockersf** [comment]
This is also part of the winit update (https://github.com/bevyengine/bevy/pull/8745/files). `LWin` will be called `SuperLeft` on the next winit version.

**Comment by alice-i-cecile** [comment]
Great, I like the doc aliases regardless of the final decision here.

File: crates/bevy_input/src/keyboard.rs
Comment (alice-i-cecile) [inline crates/bevy_input/src/keyboard.rs:333]:
```suggestion
    SuperLeft,
```

We should align with #8745 to avoid conflicts (and keep consistency with upstream).

File: crates/bevy_input/src/keyboard.rs
Comment (alice-i-cecile) [inline crates/bevy_input/src/keyboard.rs:386]:
```suggestion
    SuperRight,
```

File: crates/bevy_input/src/keyboard.rs
Comment (jim-ec) [reply crates/bevy_input/src/keyboard.rs:333]: Shouldn't we then also rename `LAlt`, `LBracket`, `LControl`, and `LShift` to `LeftAlt` etc.?

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:333]: `AltLeft`, but yes, I agree. I think that the increased clarity is worth the extra characters.

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:333]: I'm sure @mockersf has opinions though :p

Comment (jim-ec) [reply crates/bevy_input/src/keyboard.rs:333]: In that case we should also adjust the doc aliases to `WinLeft` etc.

**Comment by Vrixyz** [comment]
As author of in #8792, I saw this PR linked, I'm ok with handling the potential conflicts :)

The PR description mentions [winit's ModifiersKeys](https://github.com/rust-windowing/winit/blob/master/src/keyboard.rs#L1635) ; but as hinted in the [comments](https://github.com/bevyengine/bevy/pull/8792#issuecomment-1582794631), I think I'd prefer to keep our namings matched to [winit's KeyCode](https://github.com/rust-windowing/winit/blob/master/src/keyboard.rs#L201).

> While this differs from winit's KeyCode naming

I'm alright with temporary winit difference, if consensus is that it's clearer. If there's no issue with #8745, namings should be back to similar after we update to winit 0.29 (hopefully bevy 0.12).

File: crates/bevy_input/src/keyboard.rs
Comment (mockersf) [reply crates/bevy_input/src/keyboard.rs:333]: > I'm sure @mockersf has opinions though :p

If we change one, we should change them all... but no opinion on wether to do it now or with winit update later

**Comment by jim-ec** [comment]
So I renamed the key code to `SuperLeft` to be consistent with winit.

**Comment by MrGVSV** [comment]
Also, the title and description (at least the migration guide) should probably be updated to make the changelog work easier for 0.11

File: crates/bevy_input/src/keyboard.rs
Comment (TimJentzsch) [inline crates/bevy_input/src/keyboard.rs:322]: Very minor thing: The doc string is now the reverse of the actual name.
Not sure if this matters though.

Comment (MrGVSV) [reply crates/bevy_input/src/keyboard.rs:322]: Yeah I'm not a fan of winit's naming here but it makes sense to be consistent with them

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:322]: Yeah I think this is fine.

**Comment by alice-i-cecile** [comment]
Alright, merging this in as no one else has expressed opposition after more socialization :)

**Pull request merged**
